packages/mdc-list/foundation.js

```diff
@@ -264,8 +267,7 @@ class MDCListFoundation extends MDCFoundation {
           this.setSelectedIndex(currentIndex);
         }
 
-        // Explicitly activate links, since we're preventing default on Enter, and Space doesn't activate them.
-        this.adapter_.followHref(currentIndex);
+        this.adapter_.notifyAction(currentIndex);
```

Comment:
Related to https://github.com/material-components/material-components-web/pull/4151#pullrequestreview-189079533, I am wondering if we should do a couple of things here:

1. Call `preventDefaultEvent` in all cases rather than just the `isSingleSelectionList_` case?
2. Only call `notifyAction` if `ELEMENTS_KEY_ALLOWED_IN.indexOf(tagName) === -1`, since e.g. pressing space while a text input is focused shouldn't count?

Reply:
This seems like out of scope of this PR. Can we track this as a separate issue?

We also only call `setSelectedIndex` if `ELEMENTS_KEY_ALLOWED_IN.indexOf(tagName) === -1`

Reply:
Fixing #1

Reply:
I think we should also fix #2 because otherwise we're going to be firing an event that users may react to, in cases where it really isn't appropriate...

I don't really see how it's out of scope, when these PRs are creating these issues... these are complex changes that I'm trying to prevent new bugs from arising out of.

Reply:
I looked at 2nd point: The block of code is executed only if the keydown event is received on list root element (i.e., `if (isRootListItem)`). So the `notifyAction` will never be called if keydown received on list item child elements.

Regarding the 1st point: the scroll behaviour on space key already exists without this PR. I would still prefer addressing this issue in separate PR.

Reply:
Yeah, you're right about the 2nd point.

However, the 1st issue does not exist without this and the menu PR, because these PRs together remove the `preventDefault` logic that currently exists in Menu.

(Non-)Repro:

1. `git checkout master`
2. `git checkout list_custom_event_menu_update_issue3481 -- test/screenshot/spec/mdc-menu` to get your menu screenshot test page against master
3. `npm start`
4. Browse to http://localhost:8080/spec/mdc-menu/classes/baseline.html
5. Make sure the window size is small enough that the body has a scrollbar.
6. Use space key to select an item

Against master: Scroll doesn't happen.
On menu branch (which is based on this branch): Scroll happens.

Reply:
I think I know what is happening.

This is indeed issue created with next PR where we removed preventDefault code from menu foundation. I thought preventDefault is only intended for selectable list, I changed it to preventDefault for all list variants to avoid complexity for menu.

Reply:
I just re-tested this with the update and it looks like it's resolved now 👍